### DISCUSSION TITLE
New UR Type for Bitcoin message signing

### DISCUSSION THREAD

**Discussion by aaronisme** [BlockchainCommons/Gordian-Developer-Community#63] [Specifications]
I am wondering whether there is any plan for adding a ur type for bitcoin message signing?

**Comment by wolfmcnally** [comment]
Yes, we plan to add a way to use ur:crypto-request and ur:crypto-response to sign PSBTs. This has already been speced [here](https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2021-001-request.md#psbts). We have yet to release any implementations, but [Gordian Seed Tool](https://github.com/BlockchainCommons/GordianSeedTool-iOS) and [keytool-cli](https://github.com/BlockchainCommons/keytool-cli) will contain the first.

@ChristopherA

**Reply by aaronisme** [reply]
crypto-request seems cool, but It seems for transactions? What about the raw message like using the private key to sign "hello world"

**Comment by ChristopherA** [comment]
Yes, we want some sign content options with crypto-request. We have been puzzling through some choices of signature types in a research project at https://github.com/BlockchainCommons/musign-cli

The biggest issue is still signing multsig, or signing with p2sh. I had hoped @kallewoof’s bip322 would have more traction.

**Comment by wolfmcnally** [comment]
> I am wondering whether there is any plan for adding a ur type for bitcoin message signing?

Please see the 0.7.0 tag of keytool-cli, which now includes requests and responses for signing PSBTs. This functionality is coming to the Gordian Seed Tool app very soon.
